GafferArnold : Support multiple outputs on OSL shaders

danieldresser-ie opened the pull request:
Arnold does not yet directly support multiple outputs from OSL shaders, but it does allow us to select which output we want, so we can output multiple copies of a shader node, one for each output used.

There are some pretty ugly looking for-loops in here - I wrote it without using C++11 so that it can be backported.  If this approach looks correct, I'll backport it, then update this to C++11.

johnhaddon commented on python/GafferArnoldTest/IECoreArnoldPreviewTest/RendererTest.py:
```diff
@@ -1366,6 +1366,72 @@ def testPureOSLShaders( self ) :
+			IECoreScene.MeshPrimitive.createPlane( imath.Box2f( imath.V2f( -1 ), imath.V2f( 1 ) ) ),
+			r.attributes( IECore.CompoundObject( { "ai:surface" : network } ) )
+		)
+		del o
```
If we just don't assign to `o` in the first place, then we wouldn't have to `del o` here, since we don't use `o` for anything.

danieldresser-ie commented:
Addressed John's comments, and added a commit cleaning up for loop syntax.  Planning to squash and merge tomorrow morning after I glance over it one last time.
